crates/bootstrap_srv/src/server.rs

```diff
@@ -0,0 +1,333 @@
+//! bootstrap http server types.
+
+use std::sync::Arc;
+
+use crate::*;
+use tiny_http::*;
+
+/// An actual kitsune2_bootstrap_srv server instance.
+///
+/// This server is built to be direct, light-weight, and responsive.
+/// On the server-side, as one aspect toward accomplishing this,
+/// we are eschewing async code in favor of os thread workers.
+pub struct BootstrapSrv {
+    cont: Arc<std::sync::atomic::AtomicBool>,
+    workers: Vec<std::thread::JoinHandle<std::io::Result<()>>>,
+    addr: std::net::SocketAddr,
+}
+
+impl Drop for BootstrapSrv {
+    fn drop(&mut self) {
+        self.cont.store(false, std::sync::atomic::Ordering::SeqCst);
+        for worker in self.workers.drain(..) {
+            let _ = worker.join().expect("Failure shutting down worker thread");
+        }
+    }
+}
+
+impl BootstrapSrv {
+    /// Construct a new BootstrapSrv instance.
+    pub fn new(config: Config) -> std::io::Result<Self> {
+        let config = Arc::new(config);
+
+        // atomic flag for telling worker threads to shutdown
+        let cont = Arc::new(std::sync::atomic::AtomicBool::new(true));
+
+        // synchronization type for managing infos in spaces
+        let space_map = crate::SpaceMap::default();
+
+        // tiny_http configuration
+        let sconf = ServerConfig {
+            addr: ConfigListenAddr::IP(vec![config.listen_address]),
+            // TODO make the server able to accept TLS certificates
+            ssl: None,
+        };
+
+        // virtual-memory-like file system storage for infos
+        let store = Arc::new(crate::Store::default());
+
+        // start the actual http server
+        let server =
+            Arc::new(Server::new(sconf).map_err(std::io::Error::other)?);
+
+        // get the address that was assigned
+        let addr = server.server_addr().to_ip().expect("BadAddress");
+        println!("Listening at {:?}", addr);
+
+        // spawn our worker threads
+        let mut workers = Vec::with_capacity(config.worker_thread_count + 1);
+        for _ in 0..config.worker_thread_count {
+            let config = config.clone();
+            let cont = cont.clone();
+            let store = store.clone();
+            let server = server.clone();
+            let space_map = space_map.clone();
+            workers.push(std::thread::spawn(move || {
+                worker(config, cont, store, server, space_map)
+            }));
+        }
+
+        // also set up a worker for pruning expired infos
+        let maint_cont = cont.clone();
+        let maint_space_map = space_map.clone();
+        workers.push(std::thread::spawn(move || {
+            maint_worker(config, maint_cont, maint_space_map)
+        }));
+
+        Ok(Self {
+            cont,
+            workers,
+            addr,
+        })
+    }
+
+    /// Get the bound listinging address of this server.
+    pub fn listen_addr(&self) -> std::net::SocketAddr {
+        self.addr
+    }
+}
+
+fn maint_worker(
```

Comment:
I read this and thought it was a typo on "main_worker" -- maybe spelling it fully would be cleaer.

Reply:
Done: https://github.com/holochain/kitsune2/pull/10/commits/263d5ab3f9e742b19c46ce7a11d370205ad904ae